webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/services/reports/ExecutionBatchTest.scala

```diff
@@ -1054,4 +1115,34 @@ class ExecutionBatchTest extends Specification {
     }
   }
 
+  "performance for mergeCompareByRule" should {
+    val nbRuleInit = 12
+    val initData = buildDataForMergeCompareByRule("test", nbRuleInit, 12, 4)
+
+    val nodeList = (1 to 100).map("nodeId_" + _).toSeq
+
+    val runData = nodeList.map(buildDataForMergeCompareByRule(_, 15, 12, 5))
+
+    "init correctly" in {
+      val result = (ExecutionBatch.mergeCompareByRule _).tupled(initData)
+      result.size === nbRuleInit and
+      result.toSeq.map(x => x.compliance).map(x => x.success).sum === 576
+    }
+
+    "run fast enough" in {
+      runData.map(x =>  (ExecutionBatch.mergeCompareByRule _).tupled(x) )
+
+      val t0 = System.currentTimeMillis
+
+      for (i <- 1 to 10) {
+        val t0_0 = System.currentTimeMillis
+        runData.map(x => (ExecutionBatch.mergeCompareByRule _).tupled(x))
+        val t1_1 = System.currentTimeMillis
+        println(s"${i}th call to mergeCompareByRule for ${nodeList.size} nodes took ${t1_1-t0_0}ms")
+      }
+      val t1 = System.currentTimeMillis
+      println(s"Time to run test is ${t1-t0} ms")
+      (t1-t0) must be lessThan( 15000 ) // A bit number there, as it may run on slow systems
```

Comment:
I fear it will break on CI from time to time, and that if we put a big number, it will just be useless. 
Perhaps we should add a least a comment with some measures (to have an idea of timings next time we go there).
And still put a big value and see if it breaks CI or not.